feat: sync with main noir repo at 3416229b70

spalladino opened the pull request:
subrepo:
      subdir:   "noir"
      merged:   "3416229b70"
    upstream:
      origin:   "https://github.com/noir-lang/noir"
      branch:   "aztec-packages"
      commit:   "3416229b70"
    git-subrepo:
      version:  "0.4.6"
      origin:   "https://github.com/ingydotnet/git-subrepo"
      commit:   "110b9eb"

spalladino committed:
git subrepo commit (merge) noir

subrepo:
  subdir:   "noir"
  merged:   "62987342e7"
upstream:
  origin:   "https://github.com/noir-lang/noir"
  branch:   "aztec-packages"
  commit:   "75c83348e6"
git-subrepo:
  version:  "0.4.6"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "110b9eb"

TomAFrench reviewed (approved):
Diff looks as expected. Happy to merge assuming no sudden CI snafus

TomAFrench commented:
jinxed it!

TomAFrench commented:
https://github.com/noir-lang/noir/pull/4014

spalladino committed:
git subrepo pull (merge) --force noir

subrepo:
  subdir:   "noir"
  merged:   "3416229b70"
upstream:
  origin:   "https://github.com/noir-lang/noir"
  branch:   "aztec-packages"
  commit:   "3416229b70"
git-subrepo:
  version:  "0.4.6"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "110b9eb"

spalladino commented:
Updated to the latest aztec-packages branch in Noir to include your fix @TomAFrench
